app/controllers/declaration/TransportContainerController.scala

```diff
@@ -106,7 +106,7 @@ class TransportContainerController @Inject()(
   private def saveFirstContainer(mode: Mode, containerId: Option[String])(implicit request: JourneyRequest[AnyContent]) =
     containerId match {
       case Some(id) => updateCache(Seq(Container(id, Seq.empty))).map(_ => redirectAfterAdd(mode, id))
-      case None     => updateCache(Seq.empty).map(_ => navigator.continueTo(routes.SummaryController.displayPage(mode)))
+      case None     => updateCache(Seq.empty).map(_ => navigator.continueTo(routes.SummaryController.displayPage(Mode.Normal)))
```

Comment:
Why we always need here to have a `Mode.Normal`?

Reply:
That is a good question.   It's essentially a hack to fix a problem uncovered by one of the regression tests*.   It's what the code used to do when the user responded 'No' to the first "has containers" question but I'm not sure that it's right.

*the test was amending a failed submission.  If Mode.Amend is passed all the way through the journey to the summary page it means that the user cannot (re)submit the declaration.   We need a way to indicate that we are no longer "amending" and that we want to return to "normal" mode of operation.

This is something that we can discuss next week - possibly with Lukasz as he is currently working on something similar (back button behaviour in "Draft" mode)